std/algorithm/comparison.d

```diff
@@ -1085,15 +1085,49 @@ size_t levenshteinDistance(alias equals = "a == b", Range1, Range2)
     (Range1 s, Range2 t)
     if (isForwardRange!(Range1) && isForwardRange!(Range2))
 {
-    Levenshtein!(Range1, binaryFun!(equals), size_t) lev;
+    alias eq = binaryFun!(equals);
+
+    for (;;)
+    {
+        if (s.empty) return t.walkLength;
+        if (t.empty) return s.walkLength;
+        if (eq(s.front, t.front))
+        {
+            s.popFront();
+            t.popFront();
+            continue;
+        }
+        static if (isBidirectionalRange!(Range1) && isBidirectionalRange!(Range2))
+        {
+            if (eq(s.back, t.back))
+            {
+                s.popBack();
+                t.popBack();
+                continue;
+            }
+        }
+        break;
+    }
+
     auto slen = walkLength(s.save);
     auto tlen = walkLength(t.save);
+
+    if (slen == 1 && tlen == 1)
+    {
+        if (eq(s.front, t.front))
+            return 0;
+        else
```

Comment:
This redundant control flow ("else") we prefer to avoid. See [this funny discussion](http://www.yegor256.com/2015/01/21/if-then-throw-else.html?2015-03).

    return !eq(s.front, t.front);

should plenty cover it. If you want to clarify you're going for numbers, this is nice too:

    return eq(s.front, t.front) ? 0 : 1;

We're good to go after this. Thanks!